python/make_variable.h

```diff
@@ -114,18 +114,26 @@ auto do_init_0D(const T &value, const std::optional<T> &variance,
   return var;
 }
 
+// If we use py::array as the type of values and variances, then pybind11 only
+// accepts exactly numpy.array as input and not types that are convertible to
+// an array (e.g. list). But doing the conversion manually using .cast<py:array>
+// allows this function to work with anything that can be converted to an array.
+// The downside is that we make an extra copy if the input is not already an
+// array. But that is likely not important as lists / tuples should not contain
+// large data.
```

Comment:
So if it is a *numpy* array no extra copy is made?

Reply:
It might shallow copy the array but it does not copy the underlying buffer.